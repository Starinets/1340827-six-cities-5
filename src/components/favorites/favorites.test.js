import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Favorites from './favorites';
import {BrowserRouter as Router} from 'react-router-dom';
import {
  City,
  emptyFunction,
  offers
} from '../../test-data/test-data';

const favoriteList = [
  {
    city: City.HAMBURG,
    favorites: offers
  },
  {
    city: City.BRUSSELS,
    favorites: offers
  }
];

describe(`Component rendered correctly -> Favorites`, () => {

  it(`Favorites isn't empty`, () => {

    const renderer = new ShallowRenderer();

    const tree = renderer.render(
        <Router>
          <Favorites
            favoriteList={ favoriteList }
            onCityClick={ emptyFunction }
          />
        </Router>
    );
    expect(tree).toMatchSnapshot();
  });

  it(`Favorites is empty`, () => {

    const renderer = new ShallowRenderer();

    const tree = renderer.render(
        <Router>
          <Favorites
            favoriteList={ [] }
            onCityClick={ emptyFunction }
          />
        </Router>
    );
    expect(tree).toMatchSnapshot();
  });
});

