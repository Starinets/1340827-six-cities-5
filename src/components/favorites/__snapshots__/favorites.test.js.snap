// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component rendered correctly -> Favorites Favorites is empty 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Memo(Connect(Favorites))
    favoriteList={Array []}
    onCityClick={[Function]}
  />
</Router>
`;

exports[`Component rendered correctly -> Favorites Favorites isn't empty 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Memo(Connect(Favorites))
    favoriteList={
      Array [
        Object {
          "city": "Hamburg",
          "favorites": undefined,
        },
        Object {
          "city": "Brussels",
          "favorites": undefined,
        },
      ]
    }
    onCityClick={[Function]}
  />
</Router>
`;
